File: kattis/yin_yang_stones.py
# https://open.kattis.com/problems/yinyangstones
# 1s
# 1.7 Easy

# ChatGPT4: 47 chars. Not committed
# x=input()
# print(1*(x.count("W")==x.count("B")))

# 50 chars Top 10 place 8 (before my Ruby program was committed)
x=[*input()]
print(1*(x.count("W")==x.count("B")))

# 53 chars
# x=[*input()]
# print([0,1][x.count("W")==x.count("B")])


# 60 chars
# x=[*input()]
# print(1 if x.count("W")==x.count("B") else 0)

File: kattis/yin_yang_stones.pl
% https://open.kattis.com/problems/yinyangstones
% 1s
% 1.7 Easy

main :-
    read_line_to_codes(user_input,S),
    msort(S,Ss),clumped(Ss,[66-B,87-W]),
    (W =:= B -> T = 1 ; T = 0),writeln(T).

/*
% Compressed: 103 chars
main:-read_line_to_codes(user_input,S),msort(S,Ss),clumped(Ss,[66-B,87-W]),(W=:=B->T=1;T=0),writeln(T).

*/

/*
main :-
    read_line_to_codes(user_input,S),
    msort(S,Ss),clumped(Ss,Cl),
    writeln(cl=Cl),
    member(0'W-W,Cl),member(0'B-B,Cl),
    (W =:= B -> T = 1 ; T = 0),writeln(T).
*/